### ISSUE TITLE
grad_norm is nan and the loss is abnormal

### ISSUE THREAD

**Issue by soul10101** [LaVi-Lab/VG-LLM#21]
I was able to fine-tune the qwen2.5-vl 3B model normally. However, when we try to fine-tune the 7B model, the grad_norm is nan and the loss is abnormal.

{'loss': 1.7018, 'grad_norm': nan, 'learning_rate': 1.3513513513513515e-06, 'epoch': 0.01}

  1%|          | 10/1233 [07:54<14:12:16, 41.81s/it]
  1%|          | 11/1233 [08:34<16:24:43, 48.35s/it]
  1%|          | 12/1233 [09:16<15:42:22, 46.31s/it]
  1%|          | 13/1233 [09:56<15:04:58, 44.51s/it]
  1%|          | 14/1233 [10:38<14:46:55, 43.66s/it]
  1%|          | 15/1233 [11:18<14:26:13, 42.67s/it]
  1%|▏         | 16/1233 [11:59<14:11:31, 41.98s/it]
  1%|▏         | 17/1233 [12:39<14:00:44, 41.48s/it]
  1%|▏         | 18/1233 [13:19<13:52:53, 41.13s/it]
  2%|▏         | 19/1233 [14:00<13:47:45, 40.91s/it]
  2%|▏         | 20/1233 [14:40<13:43:48, 40.75s/it]
                                                    
{'loss': 1.1469, 'grad_norm': nan, 'learning_rate': 2.702702702702703e-06, 'epoch': 0.02}

  2%|▏         | 20/1233 [14:40<13:43:48, 40.75s/it]
  2%|▏         | 21/1233 [15:21<13:40:52, 40.64s/it]
  2%|▏         | 22/1233 [16:01<13:37:18, 40.49s/it]
  2%|▏         | 23/1233 [16:41<13:35:20, 40.43s/it]
  2%|▏         | 24/1233 [17:21<13:34:26, 40.42s/it]
  2%|▏         | 25/1233 [18:02<13:33:01, 40.38s/it]
  2%|▏         | 26/1233 [18:42<13:32:08, 40.37s/it]
  2%|▏         | 27/1233 [19:22<13:30:50, 40.34s/it]
  2%|▏         | 28/1233 [20:03<13:29:40, 40.32s/it]
  2%|▏         | 29/1233 [20:43<13:28:36, 40.30s/it]
  2%|▏         | 30/1233 [21:23<13:27:45, 40.29s/it]
                                                    
{'loss': 1.1638, 'grad_norm': nan, 'learning_rate': 4.0540540540540545e-06, 'epoch': 0.02}

  2%|▏         | 30/1233 [21:23<13:27:45, 40.29s/it]
  3%|▎         | 31/1233 [22:04<13:28:33, 40.36s/it]
  3%|▎         | 32/1233 [22:44<13:27:06, 40.32s/it]
  3%|▎         | 33/1233 [23:24<13:26:29, 40.32s/it]
  3%|▎         | 34/1233 [24:05<13:26:13, 40.34s/it]
  3%|▎         | 35/1233 [24:45<13:28:49, 40.51s/it]
  3%|▎         | 36/1233 [25:26<13:26:39, 40.43s/it]
  3%|▎         | 37/1233 [26:06<13:25:11, 40.39s/it]
  3%|▎         | 38/1233 [26:46<13:24:32, 40.40s/it]
  3%|▎         | 39/1233 [27:27<13:26:30, 40.53s/it]
  3%|▎         | 40/1233 [28:08<13:24:44, 40.47s/it]
                                                    
{'loss': 1.0222, 'grad_norm': nan, 'learning_rate': 4.999922377493727e-06, 'epoch': 0.03}

  3%|▎         | 40/1233 [28:08<13:24:44, 40.47s/it]
  3%|▎         | 41/1233 [28:48<13:23:00, 40.42s/it]
  3%|▎         | 42/1233 [29:28<13:22:05, 40.41s/it]
  3%|▎         | 43/1233 [30:09<13:20:24, 40.36s/it]
  4%|▎         | 44/1233 [30:49<13:19:54, 40.37s/it]
  4%|▎         | 45/1233 [31:29<13:19:46, 40.39s/it]
  4%|▎         | 46/1233 [32:10<13:17:43, 40.32s/it]
  4%|▍         | 47/1233 [32:50<13:16:32, 40.30s/it]
  4%|▍         | 48/1233 [33:30<13:16:24, 40.32s/it]
  4%|▍         | 49/1233 [34:10<13:15:20, 40.30s/it]
  4%|▍         | 50/1233 [34:51<13:16:38, 40.40s/it]
                                                    
{'loss': 0.9084, 'grad_norm': nan, 'learning_rate': 4.998542555915435e-06, 'epoch': 0.04}

  4%|▍         | 50/1233 [34:51<13:16:38, 40.40s/it]
  4%|▍         | 51/1233 [35:31<13:15:16, 40.37s/it]
  4%|▍         | 52/1233 [36:12<13:14:31, 40.37s/it]
  4%|▍         | 53/1233 [36:52<13:14:26, 40.40s/it]
  4%|▍         | 54/1233 [37:33<13:14:21, 40.43s/it]
  4%|▍         | 55/1233 [38:13<13:13:31, 40.42s/it]
  5%|▍         | 56/1233 [38:54<13:13:16, 40.44s/it]
  5%|▍         | 57/1233 [39:34<13:12:42, 40.44s/it]
  5%|▍         | 58/1233 [40:14<13:11:29, 40.42s/it]
  5%|▍         | 59/1233 [40:55<13:10:37, 40.41s/it]
  5%|▍         | 60/1233 [41:36<13:13:10, 40.57s/it]
                                                    
{'loss': 76.2389, 'grad_norm': nan, 'learning_rate': 4.995438885558294e-06, 'epoch': 0.05}

  5%|▍         | 60/1233 [41:36<13:13:10, 40.57s/it]
  5%|▍         | 61/1233 [42:16<13:11:20, 40.51s/it]
  5%|▌         | 62/1233 [42:57<13:10:52, 40.52s/it]
  5%|▌         | 63/1233 [43:37<13:10:04, 40.52s/it]
  5%|▌         | 64/1233 [44:17<13:08:22, 40.46s/it]
  5%|▌         | 65/1233 [44:58<13:07:19, 40.44s/it]
  5%|▌         | 66/1233 [45:38<13:06:58, 40.46s/it]
  5%|▌         | 67/1233 [46:19<13:06:00, 40.45s/it]
  6%|▌         | 68/1233 [46:59<13:05:27, 40.45s/it]
  6%|▌         | 69/1233 [47:40<13:04:36, 40.44s/it]
  6%|▌         | 70/1233 [48:20<13:03:44, 40.43s/it]
                                                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 4.990613507774006e-06, 'epoch': 0.06}

  6%|▌         | 70/1233 [48:20<13:03:44, 40.43s/it]
  6%|▌         | 71/1233 [49:01<13:03:21, 40.45s/it]
  6%|▌         | 72/1233 [49:41<13:02:21, 40.43s/it]
  6%|▌         | 73/1233 [50:22<13:04:01, 40.55s/it]
  6%|▌         | 74/1233 [51:02<13:03:45, 40.57s/it]
  6%|▌         | 75/1233 [51:43<13:02:12, 40.53s/it]
  6%|▌         | 76/1233 [52:23<13:01:16, 40.52s/it]
  6%|▌         | 77/1233 [53:04<13:00:41, 40.52s/it]
  6%|▋         | 78/1233 [53:44<12:59:34, 40.50s/it]
  6%|▋         | 79/1233 [54:25<12:58:21, 40.47s/it]
  6%|▋         | 80/1233 [55:05<12:57:19, 40.45s/it]
                                                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 4.984069751792077e-06, 'epoch': 0.06}

  6%|▋         | 80/1233 [55:05<12:57:19, 40.45s/it]
  7%|▋         | 81/1233 [55:46<12:56:55, 40.47s/it]
  7%|▋         | 82/1233 [56:26<12:56:32, 40.48s/it]
  7%|▋         | 83/1233 [57:06<12:55:24, 40.46s/it]
  7%|▋         | 84/1233 [57:47<12:55:21, 40.49s/it]
  7%|▋         | 85/1233 [58:28<12:54:30, 40.48s/it]
  7%|▋         | 86/1233 [59:08<12:54:01, 40.49s/it]
  7%|▋         | 87/1233 [59:49<12:54:00, 40.52s/it]
  7%|▋         | 88/1233 [1:00:29<12:53:37, 40.54s/it]
  7%|▋         | 89/1233 [1:01:10<12:51:56, 40.49s/it]
  7%|▋         | 90/1233 [1:01:50<12:51:38, 40.51s/it]

**Comment by NeutrinoLiu** [comment]
i would still stick to zero2 considering zero3 is kind of slow in my setup. 
And this works in my scenario for zero2 : https://github.com/deepspeedai/DeepSpeed/issues/5347

**Comment by zd11024** [comment]
Thank you for sharing!
